app/isomorphic/components/atoms/card-image/index.js

```diff
@@ -0,0 +1,71 @@
+import React, { useEffect, useState } from "react";
+import { get } from "lodash";
+import { ResponsiveImage } from "@quintype/components";
+import { shape, string, object, bool } from "prop-types";
+import { useSelector } from "react-redux";
+
+import "./card-image.m.css";
+
+export const CardImage = ({ story, isInitRow, pageType }) => {
+  const progressiveImageConfig = useSelector(state =>
+    get(state, ["qt", "config", "publisher-attributes", "progressive_image"], {})
+  );
+
+  const imagePerfObj =
+    progressiveImageConfig.is_enable && isInitRow ? progressiveImageConfig.initial_load : { size: "25vw", blur: 0 };
+  const [perfObj, setPerfObj] = useState(imagePerfObj);
+  const customStyleName = pageType !== "story-page" && !story["hero-image-s3-key"] ? "placeholder" : "";
+
+  useEffect(() => {
+    if (isInitRow && progressiveImageConfig.is_enable) {
+      // if we need progressive loading for the images in the page, isInitRow condition can be removed
+      const timer = () => {
```

Comment:
clearly this wont work. Please take care on what's returned and what isn't from a function